src/DGmethods/DGmodel.jl

```diff
@@ -4,243 +4,398 @@ struct HorizontalDirection <: Direction end
+    polyorder = polynomialorder(dg.grid)
 
-  Np = dofs_per_element(grid)
+    Np = dofs_per_element(grid)
 
-  communicate = !(isstacked(topology) &&
-                  typeof(dg.direction) <: VerticalDirection)
+    communicate = !(isstacked(topology) && typeof(dg.direction) <: VerticalDirection)
 
-  if hasmethod(update_aux!, Tuple{typeof(dg), typeof(bl), typeof(Q),
-                                  typeof(auxstate), typeof(t)})
-    update_aux!(dg, bl, Q, auxstate, t)
-  end
+    if hasmethod(
+        update_aux!,
+        Tuple{typeof(dg),typeof(bl),typeof(Q),typeof(auxstate),typeof(t)},
+    )
+        update_aux!(dg, bl, Q, auxstate, t)
+    end
 
   ########################
   # Gradient Computation #
   ########################
-  if communicate
-    MPIStateArrays.start_ghost_exchange!(Q)
-    MPIStateArrays.start_ghost_exchange!(auxstate)
-  end
-
-  if nviscstate > 0
-
-    @launch(device, threads=(Nq, Nq, Nqk), blocks=nrealelem,
-            volumeviscterms!(bl, Val(dim), Val(polyorder), dg.direction, Q.data,
-                             Qvisc.data, auxstate.data, vgeo, t, Dmat,
-                             topology.realelems))
-
     if communicate
-      MPIStateArrays.finish_ghost_recv!(Q)
-      MPIStateArrays.finish_ghost_recv!(auxstate)
+        MPIStateArrays.start_ghost_exchange!(Q)
+        MPIStateArrays.start_ghost_exchange!(auxstate)
     end
 
-    @launch(device, threads=Nfp, blocks=nrealelem,
-            faceviscterms!(bl, Val(dim), Val(polyorder), dg.direction,
-                           dg.gradnumflux, Q.data, Qvisc.data, auxstate.data,
-                           vgeo, sgeo, t, vmapM, vmapP, elemtobndy,
-                           topology.realelems))
+    if nviscstate > 0
 
-    communicate && MPIStateArrays.start_ghost_exchange!(Qvisc)
-  end
+        @launch(
+            device,
+            threads = (Nq, Nq, Nqk),
+            blocks = nrealelem,
+            volumeviscterms!(
+                bl,
+                Val(dim),
+                Val(polyorder),
+                dg.direction,
+                Q.data,
+                Qvisc.data,
+                auxstate.data,
+                vgeo,
+                t,
+                Dmat,
+                topology.realelems,
+            )
+        )
+
+        if communicate
+            MPIStateArrays.finish_ghost_recv!(Q)
+            MPIStateArrays.finish_ghost_recv!(auxstate)
+        end
+
+        @launch(
+            device,
+            threads = Nfp,
+            blocks = nrealelem,
+            faceviscterms!(
+                bl,
+                Val(dim),
+                Val(polyorder),
+                dg.direction,
+                dg.gradnumflux,
+                Q.data,
+                Qvisc.data,
+                auxstate.data,
+                vgeo,
+                sgeo,
+                t,
+                vmapM,
+                vmapP,
+                elemtobndy,
+                topology.realelems,
+            )
+        )
```

Comment:
My main reservations about the formatter is that it makes some of the function calls that used to fit on 2-3 lines take 20 lines. While this is cleaner, it also means you see less of the algorithm on your screen at once.

Reply:
No, unfortunately. Personally I feel this is the main downside as well.

Reply:
For particularly bad places there is always [skip formatting](https://github.com/domluna/JuliaFormatter.jl#skipping-formatting)